### PR TITLE
Add support for sub-title/actor/director/writer

### PR DESCRIPTION
**Pull request by ghost**
v4.2.0
- Add support for episode-name/cast/director/writer

### CODE REVIEW

File: src/PVRIptvData.cpp [hunk 1]
```diff
@@ -810,10 +818,10 @@ PVR_ERROR PVRIptvData::GetEPGForChannel(ADDON_HANDLE handle, int iChannelUid, ti
...
       tag.strPlot             = myTag.strPlot.c_str();
       tag.strOriginalTitle    = nullptr;  /* not supported */
-      tag.strCast             = nullptr;  /* not supported */
-      tag.strDirector         = nullptr;  /* not supported */
-      tag.strWriter           = nullptr;  /* not supported */
-      tag.iYear               = 0;     /* not supported */
+      tag.strCast             = myTag.strCast.c_str();
+      tag.strDirector         = myTag.strDirector.c_str();
+      tag.strWriter           = myTag.strWriter.c_str();
+      tag.iYear               = 0;        /* not supported */
```
Comment (phunkyfish) [inline src/PVRIptvData.cpp]: unrelated change

File: src/PVRIptvData.cpp [hunk 2]
```diff
@@ -825,15 +833,15 @@ PVR_ERROR PVRIptvData::GetEPGForChannel(ADDON_HANDLE handle, int iChannelUid, ti
       else
       {
         tag.iGenreType          = EPG_GENRE_USE_STRING;
-        tag.iGenreSubType       = 0;     /* not supported */
+        tag.iGenreSubType       = 0;   /* not supported */
```
Comment (phunkyfish) [inline src/PVRIptvData.cpp]: unrelated change.

File: src/PVRIptvData.cpp (hunk 1 above)
Comment (ghost) [reply src/PVRIptvData.cpp]: reverted

File: src/PVRIptvData.cpp (hunk 2 above)
Comment (ghost) [reply src/PVRIptvData.cpp]: reverted

**Comment by phunkyfish** [comment]
Looks good. Let me test quickly later today.

**Review by phunkyfish — APPROVED** [review]

**Comment by phunkyfish** [comment]
Tested and it all looks good.

**Comment by Sorien** [comment]
there are some other tags like 
```
<!ELEMENT credits (director*, actor*, writer*, adapter*, producer*,
                   composer*, editor*, presenter*, commentator*,
                   guest* )>
<!ELEMENT director    (#PCDATA)>
<!ELEMENT actor       (#PCDATA)>
<!ATTLIST actor role  CDATA #IMPLIED>
<!ELEMENT writer      (#PCDATA)>
<!ELEMENT adapter     (#PCDATA)>
<!ELEMENT producer    (#PCDATA)>
<!ELEMENT composer    (#PCDATA)>
<!ELEMENT editor      (#PCDATA)>
<!ELEMENT presenter   (#PCDATA)>
<!ELEMENT commentator (#PCDATA)>
<!ELEMENT guest       (#PCDATA)>
```
could you add them?

**Comment by phunkyfish** [comment]
The fields do not exist in kodi pvr so not possible.
